core/src/test/java/io/confluent/rest/ConnectionDurationFilterTest.java

```diff
@@ -0,0 +1,225 @@
+ *  OkHttp for HTTP/2.0 tests.
+ */
+public class ConnectionDurationFilterTest {
+
+  static Server server;
+  private static final long MAX_CONNECTION_DURATION_MS = 100;
+
+  @BeforeAll
+  public static void setUp() throws Exception {
+    MyAddressApplication application =
+        new MyAddressApplication(
+            new MyAddressConfig(
+                ImmutableMap.of(
+                    "listeners", "http://localhost:8080",
+                    "http2.enabled", "true",
+                    "max.connection.duration.ms", String.valueOf(MAX_CONNECTION_DURATION_MS)))
+        );
+    server = application.createServer();
+    server.start();
+  }
+
+  @AfterAll
+  public static void tearDown() throws Exception {
+    server.stop();
+  }
+
+  @Test
+  public void connectionDurationTest_ApacheHttpClient_Ephemeral_HTTP1() throws Exception {
+
+    CloseableHttpClient client = createPersistentClient();
+    List<String> addressList = new ArrayList<>();
+
+    // Send a few requests with a delay between consecutive requests that is longer
+    // than the connection duration threshold.
+    // Each request should succeed and the server should return a different address for each request.
+    for (int i=0; i<5; i++) {
+
+      HttpGet request = createRequest(server.getURI());
+      assertEquals("HTTP/1.1", request.getProtocolVersion().toString());
+
+      CloseableHttpResponse response = client.execute(request);
+      assertEquals(Status.OK.getStatusCode(), response.getStatusLine().getStatusCode());
+
+      String responseBody = EntityUtils.toString(response.getEntity());
+      assertTrue(responseBody.startsWith("127.0.0.1:"));
+
+      response.close();
+      addressList.add(responseBody);
+      Thread.sleep(MAX_CONNECTION_DURATION_MS + 10);
+    }
+    client.close();
+
+    // check that we actually used 5 different addresses under the hood
+    assertEquals(5, addressList.stream().distinct().count());
+  }
+
+  @Test
+  public void connectionDurationTest_ApacheHttpClient_Persistent_HTTP1() throws Exception {
+
+    CloseableHttpClient client = createEphemeralClient();
+    List<String> addressList = new ArrayList<>();
+
+    // Send a few requests with a delay between consecutive requests that is longer
+    // than the connection duration threshold.
+    // Each request should succeed and the server should return a different address for each request.
+    for (int i=0; i<5; i++) {
+
+      HttpGet request = createRequest(server.getURI());
+      assertEquals("HTTP/1.1", request.getProtocolVersion().toString());
+
+      CloseableHttpResponse response = client.execute(request);
+      assertEquals(Status.OK.getStatusCode(), response.getStatusLine().getStatusCode());
+
+      String responseBody = EntityUtils.toString(response.getEntity());
+      assertTrue(responseBody.startsWith("127.0.0.1:"));
+
+      response.close();
+      addressList.add(responseBody);
+      Thread.sleep(MAX_CONNECTION_DURATION_MS + 10);
```

Comment:
Going by the test name, is this test creating "persistent" tcp connections? So should the request fail given the connection remains idle for longer than MAX_CONNECTION_DURATION_MS